filterconfig/filterconfig.go

```diff
@@ -141,12 +141,32 @@ type Backend struct {
 
 // BackendAuth ... TODO: refactor after https://github.com/envoyproxy/ai-gateway/pull/43.
 type BackendAuth struct {
+	// Type specifies the auth type used to access the provider. Currently, only "APIKey" is supported.
+	//
+	// +kubebuilder:validation:Enum=APIKey
+	Type AuthType `yaml:"type"`
```

Comment:
this is just a Go package, not the k8s API so you can remove Type convention stuff
```suggestion
```